src/main/scanLogic/scanRunners/sastScan.ts

```diff
@@ -1,93 +1,118 @@
+import { Severity } from '../../types/severity';
+import { AppsConfigUtils } from '../../utils/appConfigUtils';
+import { Resource } from '../../utils/resource';
+import { ScanUtils } from '../../utils/scanUtils';
+import { Translators } from '../../utils/translators';
 import {
     AnalyzeIssue,
-    AnalyzerScanResponse,
-    AnalyzeScanRequest,
     AnalyzeLocation,
-    FileRegion,
-    FileLocation,
+    AnalyzeScanRequest,
+    AnalyzerScanResponse,
     CodeFlow,
+    FileLocation,
+    FileRegion,
     ScanType
 } from './analyzerModels';
-import { ConnectionManager } from '../../connect/connectionManager';
-import { AnalyzerUtils } from '../../treeDataProviders/utils/analyzerUtils';
-import { Resource } from '../../utils/resource';
-import { Severity } from '../../types/severity';
-import { Translators } from '../../utils/translators';
-import { ScanUtils } from '../../utils/scanUtils';
+import { BinaryRunner } from './binaryRunner';
 
-export interface EosScanRequest extends AnalyzeScanRequest {
+/**
+ * The request that is sent to the binary to scan Sast
+ */
+export interface SastScanRequest extends AnalyzeScanRequest {
     language: LanguageType;
     exclude_patterns: string[];
+    excluded_rules: string[];
 }
 
-export type LanguageType = 'python' | 'javascript' | 'java';
+export type LanguageType = 'python' | 'javascript' | 'typescript' | 'java';
 
-export interface EosScanResponse {
-    filesWithIssues: EosFileIssues[];
+export interface SastScanResponse {
+    filesWithIssues: SastFileIssues[];
 }
 
-export interface EosFileIssues {
+export interface SastFileIssues {
     full_path: string;
-    issues: EosIssue[];
+    issues: SastIssue[];
 }
 
-export interface EosIssue {
+export interface SastIssue {
     ruleId: string;
     severity: Severity;
     ruleName: string;
     fullDescription?: string;
-    locations: EosIssueLocation[];
+    locations: SastIssueLocation[];
 }
 
-export interface EosIssueLocation {
+export interface SastIssueLocation {
     region: FileRegion;
     threadFlows: FileLocation[][];
 }
 
-export class EosRunner extends BinaryRunner {
+export class SastRunner extends BinaryRunner {
     constructor(
         connectionManager: ConnectionManager,
         logManager: LogManager,
         binary?: Resource,
-        timeout: number = ScanUtils.ANALYZER_TIMEOUT_MILLISECS,
-        runDirectory?: string
+        timeout: number = ScanUtils.ANALYZER_TIMEOUT_MILLISECS
     ) {
-        super(connectionManager, timeout, ScanType.Eos, logManager, binary, runDirectory);
+        super(connectionManager, timeout, ScanType.Sast, logManager, binary);
+    }
+
+    /** @override */
+    protected async runBinary(
+        yamlConfigPath: string,
+        executionLogDirectory: string | undefined,
+        checkCancel: () => void,
+        responsePath: string
+    ): Promise<void> {
+        await this.executeBinary(checkCancel, ['zd', yamlConfigPath, responsePath], executionLogDirectory);
```

Comment:
According to ZD scanner 
```
Usage: scanner scan [OPTIONS] OUTPUT_PATH [CONFIG_PATH]
```
Here we are swapping the OUTPUT_PATH with the CONFIG_PATH as we pass yamlConfigPath, responsePath.

Reply:
I verified it in the AnalyzerManager and the order is correct